src/S3File.cc

```diff
@@ -114,18 +115,22 @@ int S3File::Open(const char *path, int Oflag, mode_t Mode, XrdOucEnv &env) {
 	this->s3_service_url = configured_s3_service_url;
 	this->s3_access_key = configured_s3_access_key;
 	this->s3_secret_key = configured_s3_secret_key;
-	std::string configured_s3_url_style = m_oss->getS3URLStyle();
+	this->s3_url_style = configured_s3_url_style;
 
-	// We used to query S3 here to see if the object existed, but of course
-	// if you're creating a file on upload, you don't care.
+    //This flag is not set when it's going to be a read operation
+    //so we check if the file exists in order to be able to return a 404
+    if (!Oflag) {
+        AmazonS3Download download(this->s3_service_url, this->s3_access_key,
```

Comment:
I'm inclined to agree with @nkissebe that this should probably be an `AmazonS3Head`. Your thoughts?

Reply:
I'm good with making the AmazonS3Head change. So, from a process perspective, do you reject the PR? 

I would prefer to open another issue for the test, because it's a new process, I'm not really sure what to do to satisfy it and I'm satisfied that my manual tests are sufficient for the scope of this change (i.e. no side effects that regression tests would catch).

Reply:
We'll settle in the middle; if you can update this to a HEAD, we can approve/merge the PR as is. I'll open an issue for keeping track of general unit tests

Reply:
Updated to use AWSS3head